types/time.go

```diff
@@ -2225,9 +2203,19 @@ func GetFormatType(format string) (isDuration, isDate bool) {
 			isDuration, isDate = false, false
 			break
 		}
-		if _, ok := durationTokens[token]; ok {
+		var durationTokens bool
+		switch token {
+		case "%h", "%H", "%i", "%I", "%s", "%S", "%k", "%l":
```

Comment:
Sounds like you don't need to check % prefix

Reply:
Sorry to say I do not understand what stand for.
Would you mean I could change `%h` to `h`?
Like:
```Go
swicth token[1:]{
		case "h", "H", "i", "I", "s", "S", "k", "l":
}
```

Reply:
You may use the byte comparison instead of string comparison, i haven't check the generated ASM code, you can do a benchmark.

Reply:
You may use the byte comparison instead of string comparison, i haven't check the generated ASM code, you can do a benchmark.